LibVLC Player

rocksdanister opened the pull request:
Implement LibVLC video player.

- [x] LivelyProperties customization.
- [x] Video stretch scalers.
- [x] Capture screenshot.
- [x] Loading screen (Core initialization is blocking.)
- [x] User interface.
- [x] ~Gif and Picture support~ (Vlc issue.)
- [x] ~New LivelyProperties.json config.~ Common file instead.

rocksdanister committed:
Initial commit

libVLCSharp sample

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rocksdanister/lively?pullRequest=2935) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rocksdanister/lively?pullRequest=2935) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rocksdanister/lively?pullRequest=2935) it.</sub>

rocksdanister committed:
post-processing..

- .. Mapped mpv livelyproperty to vlc range.
- Implemented looping.
- Libvlc flags.

rocksdanister committed:
Screenshot fn

rocksdanister committed:
Wallpaper scaler impl..

.. and refactoring

rocksdanister committed:
Fix aspect ratio fail if size is zero

rocksdanister committed:
Remove position changed looping impl

rocksdanister committed:
initialize blocking operation after ui is shown..

- .. to show waiting wallpaper screen.
- fix scaler not applying.
- fix scalerdropdown not impl in restorelivelyproperty fn.
- reload fn impl.
- added appicon.

rocksdanister committed:
Fix snapshot solid green image

rocksdanister committed:
Add loading screen

rocksdanister committed:
IWallpaper and client impl..

.. and some refactoring.

rocksdanister commented:
Looks like gamma filter is not working.

rocksdanister committed:
Fix scaling issue..

.. back to initializing Vlc core in the ui thread.

rocksdanister committed:
Gif support..

.. but it appears vlc has issues playing gif, leaving it as experimental.

rocksdanister committed:
Fix LivelyProperty restore not working

rocksdanister committed:
Map livelyproperties using pivot

rocksdanister committed:
Switching to manifest for dpi awareness fixes scaling issue

rocksdanister committed:
Disable gamma filter

rocksdanister committed:
Switch back  to App.config dpi awareness..

.. turns out scaler issue was visual studio debug build bug.
